cloudprem/main.tf

```diff
@@ -60,6 +66,11 @@ data "aws_availability_zones" "available" {}
 
 data "aws_caller_identity" "current" {}
 
+data "aws_vpc" "this" {
+  count = local.create_vpc ? 0 : 1
+  id = var.vpc_id
+}
```

Comment:
Odd. A `count` specified in a `data` block? I'm surprised that even works...
It's a little odd that we're "flipping" the count, so `count = 0` for `create_vpc = true`. I think the naming here might be confusing. Calling it `this` isn't particularly descriptive. Maybe something like `existing_vpc` or the like could be helpful?

Reply:
It's the only VPC which is why I use the "this" title, though we could definitely rename it. The reason we use count is because that data block should be invoked when an existing vpc is specified. If an existing VPC isn't specified and that data block was not conditional it would generate an error and fail the run (VPC of blank can't be found lol). Basically saying: if vpc exists, grab it's info, otherwise grab the info from the created vpc.

Reply:
I see.

Might it better to just parameterize then? We could pass in `create_vpc<bool>` and `existing_vpc<?vpcid>` from the config? Otherwise, we're doing some implicit conditional trickery?